### PR TITLE
feat: fetch categories, tasks, rotation cycles, and tenants

### LINKED ISSUE #252: Fetch categories, tasks, rotation cycles, and tenants
## Description

Please fetch data at once from `/api/sharehouse/:share_house_id` to display the categories, tasks, rotation cycles, and tenant information on the sharehouse edit page.

## Criteria

- [ ] Fetch and display all information from `/api/sharehouse/:share_house_id`
- [ ] If applicable, delete hard-coded code like mock data or related code

### PR DESCRIPTION
**Pull request by HrRn**
## Overview

- I have added fetching of the categories, tasks, rotation cycles, and tenants.

## Changes

- Call the `api.sharehouse.$get` method to fetch categories, tasks, rotation cycles, and tenant information.
- Update the maximum amount for the category and tenant based on the provided code.
- Add props to `RotationCycle.tsx`.

## Screenshots or videos

image

## Assignee Checklist:

- [x] The base branch is correct (no accidental merges)
- [x] The branch name follows our branch naming rules
- [x] The PR title follows our PR title rules
- [x] My code follows our coding style
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings or errors in the console

## Reviewer Checklist:

- [ ] Code readability and simplicity
- [ ] Follows best practices and coding standards
- [ ] Understandable and maintainable code

### CODE REVIEW

File: src/components/sharehouses-management/RotationCycle.tsx [hunk 1]
```diff
@@ -5,7 +5,11 @@ import { toast } from '@/components/ui/use-toast';
 import { useState } from 'react';
 import { SubmitHandler, useForm } from 'react-hook-form';
 
-export const RotationCycle = () => {
+interface RotationCycleProps {
```
Comment (yuki-92c) [inline src/components/sharehouses-management/RotationCycle.tsx]: Could you put prefix capital `I`?

**Review by yuki-92c — COMMENTED** [review]
Thank you Rina:)

File: src/components/sharehouses-management/RotationCycle.tsx (hunk 1 above)
Comment (HrRn) [reply src/components/sharehouses-management/RotationCycle.tsx]: Thank you for reviewing! I fixed it!

**Review by yuki-92c — APPROVED** [review]
LGTM🐼🐬

**Pull request merged**